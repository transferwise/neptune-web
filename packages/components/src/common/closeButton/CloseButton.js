import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { useIntl } from 'react-intl';
import { Cross as CrossIcon } from '@transferwise/icons';

import messages from './CloseButton.messages';

import './CloseButton.css';

export const CloseButton = React.forwardRef((props, ref) => {
  const intl = useIntl();
  const { onClick, className, size } = props;
  return (
    <button
      ref={ref}
      type="button"
      className={classNames(
        'np-close-button close btn-link text-no-decoration',
        { 'np-close-button--large': size === CloseButton.Size.LARGE },
        className,
      )}
      aria-label={intl.formatMessage(messages.ariaLabel)}
      onClick={onClick}
    >
      <CrossIcon size={size} />
    </button>
  );
});

CloseButton.Size = {
  SMALL: 16,
  LARGE: 24,
};

CloseButton.propTypes = {
  onClick: Types.func.isRequired,
  className: Types.string,
  size: Types.oneOf(Object.values(CloseButton.Size)),
};

CloseButton.defaultProps = {
  className: null,
  size: CloseButton.Size.LARGE,
};

export default CloseButton;
