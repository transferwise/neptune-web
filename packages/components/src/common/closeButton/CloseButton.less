.np-close-button {
  transition: color 0.15s ease-in-out;
  color: var(--color-control-accent);
  border: none;
  background: none;
  padding: 0;
  height: var(--size-16);

  &--large {
    height: var(--size-24);
  }

  &:hover {
    color: var(--color-control-accent-hover);
  }
}
