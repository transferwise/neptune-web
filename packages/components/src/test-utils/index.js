import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
