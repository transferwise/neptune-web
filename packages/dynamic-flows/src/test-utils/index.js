import React from 'react';
import { mount } from 'enzyme';
import { Provider } from '@transferwise/components';
import { act } from 'react-dom/test-utils';

const getMockFetchPromise = (status, jsonFn, delay, signal) => {
  const response = {
    status,
    json: jsonFn,
  };
  return new Promise((resolve) => {
    let aborted = false;
    if (signal) {
      signal.addEventListener('abort', () => {
        aborted = true;
      });
    }
    setTimeout(() => {
      return !aborted && resolve(response);
    }, delay);
  });
};

const wait = (t) => {
  return act(() => {
    return new Promise((resolve) => setTimeout(resolve, t));
  });
};

function mountWithProviders(component, locale = 'en-GB', messages = {}) {
  return mount(<Provider i18n={{ locale, messages }}>{component}</Provider>);
}

export { mountWithProviders as mount, getMockFetchPromise, wait };
