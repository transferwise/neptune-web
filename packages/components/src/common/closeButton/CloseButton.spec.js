import React from 'react';
import '@testing-library/jest-dom';
import { render } from '../../test-utils';

import CloseButton from '.';

const props = {
  className: 'className',
  onClick: jest.fn(),
};
describe('CloseButton', () => {
  it(`renders as expected`, () => {
    expect(render(<CloseButton {...props} />).container).toMatchSnapshot();
  });
});
