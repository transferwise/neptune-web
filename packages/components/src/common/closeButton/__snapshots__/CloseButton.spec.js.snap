// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders as expected 1`] = `
<div>
  <button
    aria-label="Close"
    class="np-close-button close btn-link text-no-decoration np-close-button--large className"
    type="button"
  >
    <span
      aria-hidden="true"
      class="tw-icon tw-icon-cross "
      role="presentation"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="24"
        width="24"
      >
        <path
          d="M12 13.414l7.293 7.293 1.414-1.414L13.414 12l7.293-7.293-1.414-1.414L12 10.586 4.707 3.293 3.293 4.707 10.586 12l-7.293 7.293 1.414 1.414L12 13.414z"
        />
      </svg>
    </span>
  </button>
</div>
`;
